mono/mini/interp/transform.c

```diff
@@ -2493,6 +2493,7 @@ interp_transform_call (TransformData *td, MonoMethod *method, MonoMethod *target
 				td->last_ins->data [1] = vt_stack_used;
 				td->last_ins->data [2] = vt_res_size;
 				td->last_ins->data [3] = save_last_error;
+				td->last_ins->data [4] = get_data_item_index (td, NULL);
```

Comment:
This is wrong because get_data_item_index is meant to be used with constant data. This means that it is implemented so that if you call it multiple times with the same key it will return the same index. You would need to add a non-shareable mode to get_data_item_index, so each calli_nat has its own cache.